parser_methods.py

```diff
@@ -50,7 +51,7 @@
 }
 
 
-URI_REGEX = r"http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+#]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+"
+URI_REGEX = r"h[tt|xx]+p[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+#]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+"
```

Comment:
```suggestion
URI_REGEX = r"h(tt|xx)p[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+#]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+"
```

not sure if we intentionally also want to match things like `httttps` or `hxxxxxp` which it does with the `+`

also `[]` is a character class selector (i.e. match any of the specified characters) in regexes, so having duplicate characters in `[]` doesn't have any effect